google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/ui/CloudDebugRunConfigurationPanel.java

```diff
@@ -43,11 +46,16 @@ public JPanel getMainPanel() {
     return panel;
   }
 
-  public String getProjectName() {
-    return elysiumProjectId.getText();
+  public ProjectSelector getProjectSelector() {
+    return projectSelector;
   }
 
-  public void setProjectName(String name) {
-    elysiumProjectId.setText(name);
+  public void triggerValidation() {
+    hiddenValidationTrigger.doClick();
+  }
+
+  private void createUIComponents() {
+    hiddenValidationTrigger = new JBCheckBox();
+    hiddenValidationTrigger.setVisible(false);
```

Comment:
Not for this PR by any means, but I wonder if instead of duplicating this "hack" everywhere, we can create a base panel class that encapsulates this..like a `HiddenValidationTriggeringPanel` or something..

Reply:
Probably or even better make that interface that `SettingsEditor` is supposed to recognize work and implement it in `ProjectSelector`. I'll spend some time debugging it and see if it can actually work.

Reply:
Yea ideally that would be the solution.